test/e2e/namespace_test.go

```diff
@@ -74,3 +81,63 @@ func TestMultipleNamespace(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+// This test is to ensure we do not leak deletion of services in other namespaces when deleting a route.
+func TestConflictingRouteService(t *testing.T) {
+	t.Parallel()
+
+	names := test.ResourceNames{
+		Config:        test.AppendRandomString("conflicting-route-service"),
+		Route:         test.AppendRandomString("conflicting-route-service"),
+		TrafficTarget: "chips",
+		Image:         pizzaPlanet1,
+	}
+
+	// Create a service in a different namespace but have the same route name
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      test.AppendRandomString("conflicting-route-service"),
+			Namespace: test.AlternativeServingNamespace,
+			Labels: map[string]string{
+				serving.RouteLabelKey: names.Route,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Type:         corev1.ServiceTypeExternalName,
+			ExternalName: "some-internal-addr",
+		},
+	}
+
+	altClients := SetupAlternativeNamespace(t)
+	altClients.KubeClient.Kube.CoreV1().Services(test.AlternativeServingNamespace).Create(svc)
+	test.CleanupOnInterrupt(func() {
+		altClients.KubeClient.Kube.CoreV1().Services(test.AlternativeServingNamespace).Delete(svc.Name, &v1.DeleteOptions{})
+	})
+	defer altClients.KubeClient.Kube.CoreV1().Services(test.AlternativeServingNamespace).Delete(svc.Name, &v1.DeleteOptions{})
+
+	clients := Setup(t)
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer test.TearDown(clients, names)
+
+	if _, err := test.CreateConfiguration(t, clients, names, &test.Options{RevisionTimeoutSeconds: revisionTimeoutSeconds}); err != nil {
```

Comment:
Could this be simplified via the usual `CreateRunLatestServiceReady`?

Reply:
Definitely simplified it, thanks!